c/cert/src/rules/ENV30-C/DoNotModifyTheReturnValueOfCertainFunctions.ql

```diff
@@ -0,0 +1,68 @@
+/**
+ * @id c/cert/do-not-modify-the-return-value-of-certain-functions
+ * @name ENV30-C: Do not modify the return value of certain functions
+ * @description Return value of getenv and similar functions cannot be modified.
+ * @kind path-problem
+ * @precision very-high
+ * @problem.severity error
+ * @tags external/cert/id/env30-c
+ *       correctness
+ *       external/cert/obligation/rule
+ */
+
+import cpp
+import codingstandards.c.cert
+import semmle.code.cpp.dataflow.DataFlow
+import DataFlow::PathGraph
+
+/*
+ * Call to functions that return pointers to environment objects that should not be modified.
+ */
+
+class NotModifiableCall extends FunctionCall {
+  NotModifiableCall() {
+    this.getTarget()
+        .hasGlobalOrStdName(["getenv", "setlocale", "localeconv", "asctime", "strerror"])
```

Comment:
not sure if this makes a big difference but why not `hasGlobalName` instead? will this FP on CPP code bases (example `std::getenv` - do we want this rule to pick that up)?